### PR TITLE
[CI] update chrome install steps for Auth builds.

### PR DESCRIPTION
**Pull request by DellaBitta**
### Discussion

Update the Chrome installation steps to use the new method of installation. This unlocks us to test against the latest and greatest chrome versions.

Changes include:
 * The use of puppeteer to install chrome on the CI VMs.
 * An update to the chromedriver npm module so that it may be properly configured to use the new Chrome download URLs.
 * An update of node from v14 to v16 for all CI runners. The latest version of Chrome Driver requires at v16+.
 * The addition of `--npm-path npm` to all `run-s` and `run-p` invocations, as they otherwise fail when invoked via `npx` on node v16.
 * Removed the installation of chrome in the `Test Auth on Firefox If Changed` in `test-changed-auth.yml` as it seemed like it seemed superfluous.

### Testing

CI

### API Changes

N/A

### CODE REVIEW

File: packages/auth/package.json [hunk 1]
```diff
@@ -124,7 +124,7 @@
     "@rollup/plugin-json": "4.1.0",
     "@rollup/plugin-strip": "2.1.0",
     "@types/express": "4.17.17",
-    "chromedriver": "98.0.1",
+    "chromedriver": "114.0.2",
```
Comment (prameshj) [inline packages/auth/package.json:127]: This pins the version to 114.0.2 for manual installs/test runs on our development machines?

**Review by prameshj — APPROVED** [review]

File: .github/workflows/test-changed-auth.yml [hunk 2]
```diff
@@ -5,32 +5,31 @@ on: pull_request
...
       - name: install Chrome stable
-        # Pin Chrome version 114.0.5735.90-1 to avoid install failures like "No such object: chromedriver/LATEST_RELEASE_115.0.5790".
-        # The failure happens because https://chromedriver.chromium.org/downloads only goes up to version 114.
-        # TODO(b/297380444) Update script to install the latest Chrome and ChromeDriver.
         run: |
-          sudo apt-get update
-          sudo apt-get install wget
-          sudo wget http://dl.google.com/linux/chrome/deb/pool/main/g/google-chrome-stable/google-chrome-stable_114.0.5735.90-1_amd64.deb
-          sudo apt-get install -f ./google-chrome-stable_114.0.5735.90-1_amd64.deb --allow-downgrades
+          npx @puppeteer/browsers install chrome@stable
```
Comment (prameshj) [inline .github/workflows/test-changed-auth.yml:23]: fyi @NhienLam

File: .github/workflows/test-changed-auth.yml [hunk 3]
```diff
@@ -49,27 +48,22 @@ jobs:
...
     steps:
       - name: install Firefox stable
         run: |
           sudo apt-get update
           sudo apt-get install firefox
           sudo apt-get install wget
-          sudo wget http://dl.google.com/linux/chrome/deb/pool/main/g/google-chrome-stable/google-chrome-stable_114.0.5735.90-1_amd64.deb
-          sudo apt-get install -f ./google-chrome-stable_114.0.5735.90-1_amd64.deb --allow-downgrades
-
+          npx @puppeteer/browsers install chrome@stable
```
Comment (hsubox76) [inline .github/workflows/test-changed-auth.yml:57]: Just out of curiosity, why does the Firefox job install Chrome stable? Will something break if Chrome isn't there?

**Review by hsubox76 — COMMENTED** [review]
Can you also amend the description to specify that the `--npm-path` option is only needed when `npx` is involved? The problem (I think) is that `npm-run-all` mistakenly uses `npx` instead of `npm` if the process was started with `npx`. This was fixed in the fork npm-run-all2 which we may want to look into switching to at some point: https://github.com/bcomnes/npm-run-all2/pull/96

File: packages/auth/package.json (hunk 1 above)
Comment (DellaBitta) [reply packages/auth/package.json:127]: Yes, of the chromedriver dependency which downloads a varying version of the chromedriver binary (depending oh the current version of chrome.)

File: .github/workflows/test-changed-auth.yml (hunk 3 above)
Comment (DellaBitta) [reply .github/workflows/test-changed-auth.yml:57]: I haven't checked this iteration, but yes, it was required previously. I can take it out and see what happens.

**Review by hsubox76 — APPROVED** [review]

**Pull request merged**